### DISCUSSION TITLE
3D temperature profile to 2D PF simulation

### DISCUSSION THREAD

**Discussion by fervaldies** [idaholab/moose#28761] [Q&A Modules: Phase field]
Hello, I was wondering if it is possible to run a thermal simulation in 3D and then extract the data of a 2D cut (or just a thin 3D slice of that model) and use those temperature values into a phase field simulation that has that exact shape and volume. I want to do a solidification model with grain growth and need the temperature evolution for that section of the model to simulate a 2D simulation (just 1 element in depth).

**Comment by GiudGiud** [comment]
Hello

There are a number of ways you could do that.
You could load the 3D results file (checkpoint or exodus) in a SolutionUO and set fields using SolutionAUx. The location of the 2D plane in the 3D system will be used as the location to sample.

You could save the variable results in the 3D simulation using ElementValueSampler vectorpostprocessor, then read this data back into the 2D simulation using a PropertyFileReader and a PiecewiseConstrantFromCSV function

**Reply by fervaldies** [reply]
Hello, first, thank you for your answer. So, I am usually using Paraview and I have seen the Slice feature and can have the region (or regions) where my blocks will be located. But how do I extract that temperature field and use it for my other simulation?

**Reply by GiudGiud** [reply]
in Paraview I would not know.
If you wanted to slice in MOOSE you could modify this vectorprocessor
https://mooseframework.inl.gov/source/vectorpostprocessors/ElementsAlongPlane.html
to also sample element values like this one
https://mooseframework.inl.gov/source/vectorpostprocessors/ElementValueSampler.html

**Reply by fervaldies** [reply]
I have tried using ElementValueSampler vectorpostprocessor, but it is creating a new CSV file for every time step, so I am not really sure how to then read each of these CSV files for each time step with the PiecewiseConstrantFromCSV function

**Reply by GiudGiud** [reply]
If you are trying to use a variable field through a function, this is not an intended use of functions. 

You should use a material property instead. These can couple in variables. Or a Functor if you code is using that
